tests/Thumbnail/LiipImagineThumbnailTest.php

```diff
@@ -28,6 +30,84 @@
+        )->willReturn('/my/private/path/default/0011/24/ASDASDAS.png');
+        $resolverRegistry = $this->prophesize(ResolverRegistryInterface::class);
+        $resolverRegistry->getResolver('mycontext_medium')->willReturn($resolver->reveal());
+
+        $thumbnail = new LiipImagineThumbnail($cacheManager, $resolverRegistry->reveal());
+
+        $filesystem = new Filesystem(new InMemory(['myfile' => 'content']));
+        $referenceFile = new File('myfile', $filesystem);
+
+        $formats = [
+            'admin' => ['height' => 50, 'width' => 50, 'quality' => 100],
+            'mycontext_medium' => ['height' => 500, 'width' => 500, 'quality' => 100],
+            'anothercontext_large' => ['height' => 500, 'width' => 500, 'quality' => 100],
+        ];
+
+        $resizer = $this->prophesize(ResizerInterface::class);
+        $resizer->resize()->willReturn(true);
+
+        $media = new Media();
+        $media->setName('ASDASDAS.png');
+        $media->setProviderReference('ASDASDAS.png');
+        $media->setId(1023456);
+        $media->setContext('default');
+
+        $provider = $this->prophesize(MediaProviderInterface::class);
+        $provider->requireThumbnails()->willReturn(true);
+        $provider->getReferenceFile()->willReturn($referenceFile);
+        $provider->getFormats()->willReturn($formats);
+        $provider->getResizer()->willReturn($resizer);
+        $provider->generatePrivateUrl()->willReturn('/my/private/path');
+        $provider->generatePublicUrl()->willReturn('/my/public/path');
+        $provider->getFilesystem()->willReturn($filesystem);
+        $provider->getReferenceImage($media)->willReturn('default/0011/24/ASDASDAS.png');
+        $provider->getCdnPath(
+            'default/0011/24/ASDASDAS.png',
+            null
+        )->willReturn('cache/media/default/0011/24/ASDASDAS.png');
+
+        $thumbnail->generate($provider->reveal(), $media);
+        $this->assertSame('default/0011/24/ASDASDAS.png', $thumbnail->generatePublicUrl(
+            $provider->reveal(),
+            $media,
+            MediaProviderInterface::FORMAT_ADMIN
+        ));
+        $this->assertSame('cache/media/default/0011/24/ASDASDAS.png', $thumbnail->generatePublicUrl(
+            $provider->reveal(),
+            $media,
+            'mycontext_medium'
+        ));
+        $this->assertSame('/my/private/path/default/0011/24/ASDASDAS.png', $thumbnail->generatePrivateUrl(
+            $provider->reveal(),
+            $media,
+            'mycontext_medium'
+        ));
+        // trigger prophecy remove assertion
+        $thumbnail->delete(
+            $provider->reveal(),
+            $media,
+            'mycontext_medium'
+        );
+    }
+
+    /**
+     * @group legacy
+     * @expectedDeprecation Using Sonata\MediaBundle\Thumbnail\LiipImagineThumbnail without a Sonata\MediaBundle\LiipImagine\ResolverRegistryInterface is deprecated since version 3.16 and will no longer be possible in 4.0.
```

Comment:
```suggestion
     * @expectedDeprecation Using Sonata\MediaBundle\Thumbnail\LiipImagineThumbnail without a Sonata\MediaBundle\LiipImagine\ResolverRegistryInterface is deprecated since sonata-project/media-bundle 3.x and will no longer be possible in 4.0.
```